monkey/monkey_island/cc/resources/propagation_credentials.py

```diff
@@ -35,7 +35,18 @@ def get(self, collection=None):
     @auth_token_required
     @roles_accepted(AccountRole.ISLAND_INTERFACE.name)
     def put(self, collection=None):
-        credentials = [Credentials(**c) for c in request.json]
+        try:
+            credentials = []
+            for credential_pair in request.json:
+                credentials.append(Credentials(**credential_pair))
+        except (TypeError, ValueError) as err:
+            return {
+                "error": {
+                    "message": f"{str(err)}",
+                    "bad_input": f"{credential_pair}"
```

Comment:
This could potentially lead to a leak in sensitive information. I would advocate we give the user the index of the malformed credentials and not the credentials themselves. The tests should be updated to test that the correct index is returned.

Reply:
Would the index be the same as what we have in the UI?